src/OpenLoco/src/MessageManager.cpp

```diff
@@ -217,7 +217,7 @@ namespace OpenLoco::MessageManager
                 auto* town = TownManager::get(static_cast<TownId>(message.itemSubjects[2] & 0xFF));
                 args.push(town->name);
 
-                const auto formatStr = cargoObj->flags & CargoObjectFlags::unk2 ? StringIds::message_first_string_delivery_arives_at : StringIds::message_first_string_arrive_at;
+                const auto formatStr = ((cargoObj->flags & CargoObjectFlags::unk2) != CargoObjectFlags::none) ? StringIds::message_first_string_delivery_arives_at : StringIds::message_first_string_arrive_at;
```

Comment:
```suggestion
                const auto isDelievering = cargoObj->flags & CargoObjectFlags::unk2) != CargoObjectFlags::none;
                const auto formatStr = isDelievering ? StringIds::message_first_string_delivery_arives_at : StringIds::message_first_string_arrive_at;
```
I think this makes it a bit easier to read

Reply:
Looking at this further. Do we know what "unk2" means?  If that means it is delivering, can we just rename the enum to "delivering"? If so, we could combine that with the hasFlags function I just added and it could read as follows:

const auto formatStr = cargoObj->hasFlags(CargoObjectFlags::delivering) ?StringIds::message_first_string_delivery_arives_at : StringIds::message_first_string_arrive_at;